controller/backup_target_controller.go

```diff
@@ -367,6 +430,43 @@ func (btc *BackupTargetController) reconcile(name string) (err error) {
 		return err
 	}
 
+	// Check if it is the default backup target or not.
+	backupTargetMap, err := btc.ds.ListBackupTargets()
+	if err != nil {
+		log.WithError(err).Errorf("Failed to get all backup targets")
+		return err
+	}
+	if len(backupTargetMap) == 1 {
+		if _, ok := backupTargetMap[name]; ok {
+			backupTarget.Spec.Default = true
+			backupTarget.Status.Default = true
+		}
+	} else {
+		if backupTarget.Spec.Default {
+			for _, btObj := range backupTargetMap {
+				if btObj.Status.Default && btObj.Name != name {
+					btObj.Status.Default = false
+					if _, err := btc.ds.UpdateBackupTargetStatus(btObj); err != nil {
+						log.WithError(err).Errorf("Failed to update backup target %s status", btObj.Name)
+						return err
+					}
+					updatedBtObj, err := btc.ds.GetBackupTarget(btObj.Name)
```

Comment:
Do we need to re-get the obj or we can use the obj returned by `UpdateBackupTargetStatus`
We reuse the obj above and use it to update spec [L374](https://github.com/longhorn/longhorn-manager/pull/2182/files#diff-b881cb9d9886c8b07c3081dc101c18685f066469dee34f80bcdb7b98f76acacdR374)

Reply:
Yes, modified to use the obj returned by UpdateBackupTargetStatus. Thanks.